alibi_detect/utils/keops/kernels.py

```diff
@@ -1,10 +1,10 @@
 from pykeops.torch import LazyTensor
 import torch
 import torch.nn as nn
-from typing import Callable, Optional
+from typing import Callable, Optional, Union
 
 
-def sigma_mean(x: LazyTensor, y: LazyTensor, dist: LazyTensor, n_min: int = None) -> torch.Tensor:
+def sigma_mean(x: LazyTensor, y: LazyTensor, dist: LazyTensor, n_min: int = 100) -> torch.Tensor:
```

Comment:
What's the reason for changing the default here?

Reply:
Inferring the bandwidth requires `[n_min, n_min]` matrix to be evaluated just to check whether the passed `x == y`. Initially it defaulted to `nx`. This wasn't as big a deal for the normal MMD as by default you do this once at init, so it doesn't impact the prediction time. However for the learned kernel MMD you would have to do this every single prediction call as you learn new kernels, and this becomes a bottleneck, even though it is just a simple check in case you haven't provided an initial bandwidth (which will be updated during learning anyway afterwards). As a result, I reduced it to a default value of `min(nx, n_min)` with `n_min=100` as default setting. I believe this is a reasonable check to see whether the instances in x and y are the same (so you basically check the first `min(nx, n_min)` instances to see if they are the same).

Reply:
Just thinking that as an alternative I could infer the bandwidth by default in the same way as for MMD, and provide that bandwidth as initial one for every new kernel. This might actually be better as it aligns closer with normal MMD.

Reply:
Actually ignore the above comment. The deep kernel is defined outside of the detector so we can't assume access to heuristic setting.